src/app/Scenes/PartnerOffer/PartnerOfferContainer.tsx

```diff
@@ -0,0 +1,51 @@
+import { Flex } from "@artsy/palette-mobile"
+import { usePartnerOfferCheckoutMutation$data } from "__generated__/usePartnerOfferCheckoutMutation.graphql"
+import { LoadingSpinner } from "app/Components/Modals/LoadingModal"
+import { usePartnerOfferMutation } from "app/Scenes/PartnerOffer/mutations/usePartnerOfferCheckoutMutation"
+import { goBack, navigate } from "app/system/navigation/navigate"
+import { useEffect } from "react"
+
+export const PartnerOfferContainer: React.FC<{ partnerOfferID: string }> = ({ partnerOfferID }) => {
+  const handleRedirect = (response: usePartnerOfferCheckoutMutation$data) => {
+    const orderOrError = response.commerceCreatePartnerOfferOrder?.orderOrError
+
+    if (orderOrError?.error) {
+      const { code, data } = orderOrError.error
+      const artworkId = JSON.parse(data?.toString() ?? "")?.artwork_id
+
+      if (code === "expired_partner_offer") {
+        // TODO: add params to show expired banner (EMI-1606)
+        navigate(`/artwork/${artworkId}`, { replaceActiveScreen: true })
+        return
+      }
+
+      if (code === "not_acquireable") {
+        // TODO: add params to show unavailable/sold banner (EMI-1606)
+        navigate(`/artwork/${artworkId}`, { replaceActiveScreen: true })
+        return
+      }
+    }
```

Comment:
I think we might need to throw a return here in case the error code is not in the list above or have an else statement here to make sure we don't go to the next block if we have a new error that is not supported in the app. Something like this 👇 
```suggestion
    } else {
```